UML editor/src/test/java/UMLClassTest.java

```diff
@@ -18,157 +20,180 @@ public class UMLClassTest {
-        List<Object> parameters = UMLMethods.classMethods.get("TestClass1").get("method1");
-        assertFalse(parameters.contains("String param2"), "Parameter should be removed from method");
-    }
+        UMLClassInfo classInfo = UMLClass.classMap.get("TestClass1");
+        UMLMethodInfo method = classInfo.getMethodByName("method1");
+        assertFalse(method.getParameters().contains("String param2"), "Parameter should be removed from method");
+        assertTrue(method.getParameters().contains("int param1"), "Parameter should not be removed from method");
 
+    }
 
     @Test
     public void testChangeParameters() {
         UMLClass.addClass("TestClass1");
         methodsManager.addMethod("TestClass1", "method1", Arrays.asList("int param1"));
-        methodsManager.changeParameters("TestClass1", "method1", Arrays.asList("String newParam"));
-        List<Object> parameters = UMLMethods.classMethods.get("TestClass1").get("method1");
-        assertEquals(1, parameters.size(), "Parameters should be replaced");
-        assertTrue(parameters.contains("String newParam"), "New parameter should be set");
+        methodsManager.changeParameters("TestClass1", "method1", Arrays.asList("String param1", "boolean param2"));
+        UMLClassInfo classInfo = UMLClass.classMap.get("TestClass1");
+        UMLMethodInfo method = classInfo.getMethodByName("method1");
+        assertEquals(2, method.getParameters().size(), "Parameters should be updated");
+        assertTrue(method.getParameters().contains("String param1"), "New parameters should be set");
+        assertTrue(method.getParameters().contains("boolean param2"), "New parameters should be set");
+    }
 
+    @Test
+    public void testAddClass() {
+        UMLClass.addClass("TestClass1");
+        assertTrue(UMLClass.classMap.containsKey("TestClass1"), "Class should be added to the map");
     }
 
     @Test
-    public void testListClassesWithDetails() {
+    public void testAddDuplicateClass() {
         UMLClass.addClass("TestClass1");
-        fieldsManager.addField("TestClass1", "field1");
-        methodsManager.addMethod("TestClass1", "method1", Arrays.asList("int param1"));
-        UMLClass.addClass("TestClass2");
-        fieldsManager.addField("TestClass2", "int 5");
-        methodsManager.addMethod("TestClass2", "method2", Arrays.asList("String param2"));
-        UMLClass.listClasses();
+        UMLClass.addClass("TestClass1");
+        assertEquals(1, UMLClass.classMap.size(), "Duplicate class should not be added");
+    }
+
+    @Test
+    public void testRemoveClass() {
+        UMLClass.addClass("TestClass1");
+        UMLClass.removeClass("TestClass1");
+        assertFalse(UMLClass.classMap.containsKey("TestClass1"), "Class should be removed from the map");
+    }
+
+    @Test
+    public void testRenameClass() {
+        UMLClass.addClass("TestClass1");
+        UMLClass.renameClass("TestClass1", "RenamedClass");
```

Comment:
Clear and descriptive tests, and good use of assertions. We also get good testing on edge cases